app/javascript/components/top-navbar/user-options.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { Dropdown, Icon, MenuItem } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import {
+  groupProps, recursiveUserMenuProps, userMenuProps,
+} from './recursive-props';
+
+
+const UserOptions = ({
+  currentUser, applianceName, miqGroups, currentGroup, userMenu,
+}) => (
+  <Dropdown
+    id="dropdownMenu2"
+    componentClass={
+      ({ children, ...props }) => (
+        <li
+          {...props}
+          className="dropdown"
+        >
+          {children}
+        </li>
+      )}
+  >
+    <Dropdown.Toggle
+      componentClass={
+        ({ children, ...props }) => (
+          <a
+            {...props}
+            aria-haspopup="true"
+            data-toggle="dropdown"
+            className="dropdown-toggle nav-item-iconic"
+          >
+            {children}
+          </a>
+        )
+      }
+    >
+      <Icon type="pf" name="user" />
+      <p id="username_display" data-userid={currentUser.userid} className="navbar__user-name">{`${currentUser.name} | ${applianceName}`}</p>
+    </Dropdown.Toggle>
+    <Dropdown.Menu>
+      <MenuItem disabled>{sprintf(__('Server: %s'), applianceName)}</MenuItem>
+      { miqGroups.length > 1 ? (
+        <li id="droptownMenuGroup" className="dropdown-submenu pull-left">
+          <Dropdown.Toggle
+            useAnchor
+            noCaret
+            componentClass={
+              ({ children }) => (
+                <a href="#">
+                  {children}
+                </a>
+              )
+            }
+          >
+                Change Group
+          </Dropdown.Toggle>
+          <Dropdown.Menu className="scrollable-menu">
+            {miqGroups.sort().map(group => (
+              (currentGroup).id === group.id ? (
+                <MenuItem id="current-group" key={group.id} disabled title={__('Currently Selected Group')}>
+                  {`${group.description} (${__('Current Group')})`}
+                </MenuItem>
+              ) : (
+                <MenuItem
+                  id={group.description}
+                  key={group.id}
+                  title={__('Change to this Group')}
+                  onClick={(e) => {
+                    e.preventDefault();
+                    miqChangeGroup(group.id.toString());
```

Comment:
This bit won't work, `group.id` must already be a string => https://github.com/ManageIQ/manageiq-ui-classic/pull/6197/files#r349078197